wgpu-core/src/command/render.rs

```diff
@@ -1164,17 +1161,23 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
 
                     state.index.pipeline_format = pipeline.index_format;
 
+                    let vertex_strides_len = pipeline.vertex_strides.len();
+                    let empty_slots = vertex_strides_len.saturating_sub(state.vertex.inputs.len());
+                    state
+                        .vertex
+                        .inputs
+                        .extend(iter::repeat(VertexBufferState::EMPTY).take(empty_slots));
                     // Update vertex buffer limits
                     for (vbs, &(stride, rate)) in
                         state.vertex.inputs.iter_mut().zip(&pipeline.vertex_strides)
                     {
                         vbs.stride = stride;
                         vbs.rate = rate;
                     }
-                    let vertex_strides_len = pipeline.vertex_strides.len();
                     for vbs in state.vertex.inputs.iter_mut().skip(vertex_strides_len) {
                         vbs.stride = 0;
                         vbs.rate = InputStepMode::Vertex;
+                        vbs.total_size = !0;
```

Comment:
why are we changing the total size here?

Reply:
So that unused buffers don't cause validation to fail

Reply:
but we are resetting the `stride`, so the total size wouldn't be taken into account, anyway